neo/rawio/plexon2rawio/plexon2rawio.py

```diff
@@ -0,0 +1,498 @@
+        channel_header = self.header['spike_channels'][spike_channel_index]
+        channel_name, channel_unit_id = channel_header['name'].split('.')
+        channel_unit_id = int(channel_unit_id)
+
+        spike_timestamps, unit_ids, waveforms = self._spike_channel_cache[channel_name]
+
+        if t_start is not None or t_stop is not None:
+            # restrict spikes to given limits (in seconds)
+            timestamp_frequency = self.pl2reader.pl2_file_info.m_TimestampFrequency
+            lim0 = int(t_start * timestamp_frequency)
+            lim1 = int(t_stop * self.pl2reader.pl2_file_info.m_TimestampFrequency)
+
+            # limits are with respect to segment t_start and not to time 0
+            lim0 -= self.pl2reader.pl2_file_info.m_StartRecordingTime
+            lim1 -= self.pl2reader.pl2_file_info.m_StartRecordingTime
+
+            time_mask = (spike_timestamps >= lim0) & (spike_timestamps <= lim1)
+        else:
+            time_mask = slice(None, None)
+
+        unit_mask = unit_ids[time_mask] == channel_unit_id
+        spike_timestamps = spike_timestamps[time_mask][unit_mask]
+
+        # spike timestamps are counted from the session start recording time
+        spike_timestamps += self.pl2reader.pl2_file_info.m_StartRecordingTime
+
+        return spike_timestamps
+
+    def _rescale_spike_timestamp(self, spike_timestamps, dtype):
+        # must rescale to second a particular spike_timestamps
+        # with a fixed dtype so the user can choose the precision they want
+        spike_times = spike_timestamps.astype(dtype)
+        spike_times /= self.pl2reader.pl2_file_info.m_TimestampFrequency
+        return spike_times
+
+    def _get_spike_raw_waveforms(self, block_index, seg_index, spike_channel_index,
+                                 t_start, t_stop):
+        # this must return a 3D numpy array (nb_spike, nb_channel, nb_sample)
+        # in the original dtype
+        # this must be as fast as possible.
+        # the same clip t_start/t_start must be used in _spike_timestamps()
+
+        channel_header = self.header['spike_channels'][spike_channel_index]
+        channel_name, channel_unit_id = channel_header['name'].split('.')
+
+        spike_timestamps, unit_ids, waveforms = self._spike_channel_cache[channel_name]
+
+        if t_start is not None or t_stop is not None:
+            # restrict spikes to given limits (in seconds)
+            timestamp_frequency = self.pl2reader.pl2_file_info.m_TimestampFrequency
+            lim0 = int(t_start * timestamp_frequency)
+            lim1 = int(t_stop * self.pl2reader.pl2_file_info.m_TimestampFrequency)
+            time_mask = (spike_timestamps >= lim0) & (spike_timestamps <= lim1)
+
+            # limits are with respect to segment t_start and not to time 0
+            lim0 -= self.pl2reader.pl2_file_info.m_StartRecordingTime
+            lim1 -= self.pl2reader.pl2_file_info.m_StartRecordingTime
+        else:
+            time_mask = slice(None, None)
```

Comment:
To be safe here if we correct a bug one day I would share this piece of code with the one in _get_spike_timestamps